### ISSUE TITLE
Create advanced crew search controls

### ISSUE THREAD

**Issue by ineffyble** [stt-datacore/website#178]
We've continued to expand ways you can search and filter crew, but the single text field can be confusing or unintuitive.

A few separate controls, shown if user clicks "Advanced Search", could go a long way to making it easier.

For example, have a section for "traits" which uses [tags type input](https://jsfiddle.net/agaust/p377zxu4/). Have a section for skills that lets the user tick "has MED" or select which primary skill to search for, etc etc.

**Comment by ylluminate** [comment]
It would be exceptionally helpful here if you could, while in Player Tools (https://datacore.app/playertools) -> Crew, use the search field to sort by tier, for example: 

If you type in "Tier 2" it should find all matching crew that are within that tier under the "Level __, Tier __" attribute field:

![image sample](https://p153.p0.n0.cdn.getcloudapp.com/items/bLuAqQEZ/1f84f0d7-28d2-480e-a1cd-f77ffab15b01.png?v=189791a8ed7571929ad9a3ab0d2fac15)

[Short screencast](https://p153.p0.n0.cdn.getcloudapp.com/items/yAuDyBpo/7287c0b7-6ad5-445b-a9e1-689a5b7e412a.gif?v=7ab1faad84886bfd01e56b8981a84116):
![screencast example](https://p153.p0.n0.cdn.getcloudapp.com/items/yAuDyBpo/7287c0b7-6ad5-445b-a9e1-689a5b7e412a.gif?v=7ab1faad84886bfd01e56b8981a84116)

**Comment by ineffyble** [comment]
You can sort by tier by clicking the name column multiple times.

**Comment by ylluminate** [comment]
I'm sorry @ineffyble, I'm not seeing how this solves the problem: I'm trying to illustrate that having the ability to search for only "Tier 2" for example as I have above, should filter the text that is there in the resultant rows below. This would make it possible, as a use case, to show only the crew that one has that is at "Tier 2" and then share this result with a fleet for discussion or to allow you to make other interesting matches.

**Comment by ineffyble** [comment]
@ylluminate ah okay, your first comment said "sort" not "filter" which led me down wrong path

**Comment by ylluminate** [comment]
Right, gotcha, thanks and I've replaced that term to indicate the appropriate function. Maybe "Search..." should also be renamed "Search / Filter..." instead of just "Search..." too.

**Comment by ylluminate** [comment]
Another thing that would be useful is if filtering/search could be added to the "Crew Retrieval" tab functionality. That would, with the above filtering option for other fields, be incredibly useful.
